src/common.toit

```diff
@@ -92,3 +91,248 @@ abstract class Canvas:
     return OVERLAP
 
   abstract composit frame_opacity frame_canvas/Canvas painting_opacity painting_canvas/Canvas
+
+TRANSFORM_IDENTITY_ ::= Transform.with_ [1, 0, 0, 1, 0, 0]
+TRANSFORM_90_ ::= Transform.with_ [0, -1, 1, 0, 0, 0]
+TRANSFORM_180_ ::= Transform.with_ [-1, 0, 0, -1, 0, 0]
+TRANSFORM_270_ ::= Transform.with_ [0, 1, -1, 0, 0, 0]
+
+// Classic 3x3 matrix for 2D transformations.  Right column is always 0 0 1, so
+// we don't need to store it.  We don't support scaling so the top left 2x2
+// block is always 0s, 1s, and -1s.
+class Transform:
+  array_ ::= ?  // 6-element integer array.
```

Comment:
In a separate PR: explain the order of the elements. It looks like it's the concatenated rows.